app/models/spree/payment.rb

```diff
@@ -151,7 +163,7 @@ def revoke_adjustment_eligibility
     end
 
     def validate_source
-      if source && !source.valid?
+      if source && !skip_source_validation && !source.valid?
```

Comment:
I know you only added a variable here :-)
I wonder if this is not easier and equal if written like:
```ruby
def validate_source
   return true if !source || skip_source_validation || source.valid?

   source.errors.each do |field, error|
     field_name = I18n.t("activerecord.attributes.#{source.class.to_s.underscore}.#{field}")
     errors.add(Spree.t(source.class.to_s.demodulize.underscore), "#{field_name} #{error}")
    end
    false
end

```

Reply:
ok, I gave it a try and then I thought. Wait and see if someone mentions it :joy: